### DISCUSSION TITLE
Expose defaults from _config to users

### DISCUSSION THREAD

**Discussion by Bluenix2** [encode/httpx#2223] [Ideas]
I am using HTTPX inside of a wrapper class and I want to expose some of the configuration options HTTPX has (timeouts). The issue is that I would like to retain the defaults that HTTPX uses.

Python does not make it easy to use a functions defaults when calling it unless you know at compile-time which parameters will be passes. The only option is dynamically building a dictionary and unpacking it - not a nice solution.

Therefore I think that HTTPX should expose the following constants used as defaults:
https://github.com/encode/httpx/blob/6f31bc4bea7bbd03260af6c226b24c541a203130/httpx/_config.py#L361-L363

I have two options currently: either copy in the code above, or import directly from the file. Neither of which is perfectly satisfying.

**Comment by acostapazo** [comment]
It will be fantastic to have something to configure global defaults.

This is my proposal using envars.

```python
import os

DEFAULT_TIMEOUT_CONFIG = Timeout(timeout=float(os.getenv("HTTPX_DEFAULT_TIMEOUT", 5.0)))
DEFAULT_LIMITS = Limits(
    max_connections= int(os.getenv("HTTPX_DEFAULT_LIMITS_MAX_CONNECTIONS", 100)),
    max_keepalive_connections=int(os.getenv("HTTPX_DEFAULT_LIMITS_MAX_KEEPALIVE_CONNECTIONS", 20))
) 
DEFAULT_MAX_REDIRECTS = int(os.getenv("HTTPX_DEFAULT_MAX_REDIRECTS", 20)) 
```
